warehouse/accumulo-extensions/src/main/java/datawave/ingest/table/volumeChoosers/ShardedTableDateBasedTieredVolumeChooser.java

```diff
@@ -0,0 +1,117 @@
+    private static final String PROPERTY_PREFIX = "volume.tiered.";
+    private static final String VOLUME_SUFFIX = ".volumes";
+    private static final String DAYS_BACK_SUFFIX = ".days.back";
+    private static final String DATE_PATTERN = "yyyyMMdd";
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern(DATE_PATTERN);
+
+    private static Pattern SHARD_PATTERN = Pattern.compile("\\d{8}_\\d+");
+    
+    @Override
+    public String choose(VolumeChooserEnvironment env, Set<String> options) {
+
+        if (!env.getTable().isPresent() || !env.getChooserScope().equals(VolumeChooserEnvironment.Scope.TABLE))
+            return super.choose(env, options);
+        else {
+            TableId tableId = env.getTable().get();
+            ServiceEnvironment.Configuration tableConfig = env.getServiceEnv().getConfiguration(tableId);
+            // Get variables
+            log.trace("Determining tier names using property {} for Table id: {}", Property.TABLE_ARBITRARY_PROP_PREFIX + TIER_NAMES_SUFFIX, tableId);
+            String configuredTiers = tableConfig.getTableCustom(TIER_NAMES_SUFFIX);
+            TreeMap<Long, Set<String>> daysToVolumes = getTiers(tableId, tableConfig, options, configuredTiers);
+
+            Text endRow = env.getEndRow();
+
+            Long floorKey= daysToVolumes.ceilingKey(0L);
+            if(endRow == null)
+            {
+                //this is the default tablet. No shard means this is new data
+                options = floorKey == null ? options : daysToVolumes.get(floorKey);
+
+            }else {
+                String endRowString = endRow.toString();
+                if (SHARD_PATTERN.matcher(endRowString).matches()) {
+                    String date = endRowString.substring(0, 8);
+                    LocalDate rowDate = LocalDate.parse(date, FORMATTER);
+                    LocalDate today = LocalDate.now();
+                    long days = DAYS.between(rowDate, today);
+                    floorKey = daysToVolumes.floorKey(days > 0L ? days : 0L);
+                    options = floorKey == null ? options : daysToVolumes.get(floorKey);
+                } else {
+                    log.warn("endRow does not match pattern. Is this table sharded? endRow was " + endRowString + " and tableId is " + tableId);
+                }
+            }
+            return super.choose(env, options);
+        }
+    }
+    
+    private TreeMap<Long,Set<String>> getTiers(TableId tableId, ServiceEnvironment.Configuration tableConfig, Set<String> options, String configuredTiers) {
+        TreeMap<Long,Set<String>> daysToVolumes = new TreeMap<>();
+        daysToVolumes.put(Long.MAX_VALUE, options);
+        for (String tier : StringUtils.split(configuredTiers, ',')) {
+            log.debug("Determining volumes for tier {} using property {} for Table id: {}", tier, Property.TABLE_ARBITRARY_PROP_PREFIX + PROPERTY_PREFIX + tier
```

Comment:
Functionality looks good. This method is a little dense. I'm gonna approve but just in case your interested, we could make it a bit more readable by offloading some of this stuff to helper methods:

```
Set<String> tierNames = getTierNames(tableConfig);
...
for (String tier : tierNames) {
   Set<String> volumesForCurrentTier = getVolumesForTier(tableConfig, tier);
   long daysBackForCurrentTier = getDaysBackForTier(tableConfig, tier);
```

The building of the property names is a bit dense too. Those could be shoved into methods as well:

```
  throw new IllegalArgumentException("Volumes list empty for tier " + tier + ". Ensure propety " + tierVolumesProp(tier) + " is set.");
```